search/query_language_whitebox_test.go

```diff
@@ -270,7 +271,76 @@ func TestParseMap(t *testing.T) {
 		assert.Equal(t, expectedQuery, actualQuery)
 	})
 
+	t.Run(OPTS, func(t *testing.T) {
+		t.Parallel()
+		// given
+		input := fmt.Sprintf(`{"%s": [ {"parent-exists": true}, {"tree-view": true}]}`, OPTS)
+		// Parsing/Unmarshalling JSON encoding/json
+		fm := map[string]interface{}{}
+		err := json.Unmarshal([]byte(input), &fm)
+		require.Nil(t, err)
+		// when
+		actualQuery := Query{}
+		parseOptions(fm, &actualQuery)
+		// then
+		qo := QueryOptions{ParentExists: true, TreeView: true}
+		expectedQuery := Query{Options: &qo}
+		assert.Equal(t, expectedQuery, actualQuery)
+	})
+	t.Run(OPTS+" complex query", func(t *testing.T) {
+		t.Parallel()
+		// given
+		input := fmt.Sprintf(`{"$AND":[{"title":"some"},{"state":"new"}],"%s": [ {"parent-exists": true}, {"tree-view": true}]}`, OPTS)
```

Comment:
If you use Sprintf for the OPTS you should also use it for the AND.

Reply:
And cool, that you finally have a decent query example btw. ! +1

Reply:
Done. 3b7714616807204e908ff9d68fa4d1a072d17a1d